frontend/src/component/common/ConstraintAccordion/ConstraintAccordionEdit/ConstraintAccordionEditBody/DateSingleValue/DateSingleValue.tsx

```diff
@@ -1,41 +1,116 @@
+
+    const localTimezone = useMemo<ITimezoneSelect | undefined>(() => {
+        return timezones.find(t => t.key === local);
+    }, [timezones, local]);
+
+    useEffect(() => {
+        if (pickedDate && timezone) {
+            const utc = zonedTimeToUtc(pickedDate!, timezone);
+            setValue(utc.toISOString());
+            const localDate = utcToZonedTime(utc, local);
+            setLocalDate(localDate.toISOString());
+        }
+    }, [local, pickedDate, setValue, timezone]);
+
     if (!value) return null;
 
     return (
         <>
             <ConstraintFormHeader>Select a date</ConstraintFormHeader>
-            <Input
-                id="date"
-                label="Date"
-                type="datetime-local"
-                value={parseDateValue(value)}
-                onChange={e => {
-                    setError('');
-                    const parsedDate = parseValidDate(e.target.value);
-                    const dateString = parsedDate?.toISOString();
-                    dateString && setValue(dateString);
-                }}
-                InputLabelProps={{
-                    shrink: true,
-                }}
-                error={Boolean(error)}
-                errorText={error}
-                required
+            <StyledWrapper>
+                <Input
+                    id="date"
+                    label="Date"
+                    type="datetime-local"
+                    value={parseDateValue(pickedDate)}
+                    onChange={e => {
+                        setError('');
+                        const parsedDate = parseValidDate(e.target.value);
+                        const dateString = parsedDate?.toISOString();
+                        dateString && setPickedDate(dateString);
+                    }}
+                    InputLabelProps={{
+                        shrink: true,
+                    }}
+                    error={Boolean(error)}
+                    errorText={error}
+                    required
+                />
+                <ConditionallyRender
+                    condition={showSelectableTimezone}
+                    show={
+                        <ConditionallyRender
+                            condition={timezones.length > 0}
+                            show={
+                                <GeneralSelect
+                                    options={timezones}
+                                    label={'Timezone'}
+                                    onChange={setTimezone}
+                                    value={timezone}
+                                    sx={{ minWidth: 180 }}
+                                />
+                            }
+                        />
+                    }
+                    elseShow={
+                        <p>
+                            {localTimezone?.key}{' '}
+                            {`(UTC ${localTimezone?.utcOffset})`}
```

Comment:
What happens if `localTimezone` is undefined here? Do we just show an empty `p` tag? Would it make sense to show something else if it _is_ undefined?